src/dlnodesession.cpp

```diff
@@ -0,0 +1,294 @@
+    modelVersion(modelVersion) {}
+
+DLNodeSession::~DLNodeSession() = default;
+
+void DLNodeSession::clearInputs() {
+    this->inputHandler->clearInputs();
+}
+
+ModelInstance& DLNodeSession::getModelInstance() {
+    return *this->model;
+}
+
+InferenceEngine::InferRequest& DLNodeSession::getInferRequest(const uint microseconds) {
+    auto& inferRequestsQueue = this->model->getInferRequestsQueue();
+    auto streamIdOpt = this->nodeStreamIdGuard->tryGetId(microseconds);
+    if (!streamIdOpt) {
+        SPDLOG_LOGGER_ERROR(dag_executor_logger, "Failed to get streamId on already executed node: {} session: {}", getName(), getSessionKey());
+        throw std::logic_error("Stream id is empty on already executed node");
+    }
+    return inferRequestsQueue.getInferRequest(streamIdOpt.value());
+}
+Status DLNodeSession::requestExecuteRequiredResources() {
+    Status status = StatusCode::OK;
+    ModelManager& manager = const_cast<ModelManager&>(modelManager);
+    status = manager.getModelInstance(
+        modelName,
+        modelVersion,
+        this->model,
+        this->modelUnloadGuard);
+
+    if (!status.ok()) {
+        SPDLOG_LOGGER_DEBUG(dag_executor_logger, "Getting modelInstance failed for node: {} session: {} with: {}", getName(), getSessionKey(), status.string());
+        return status;
+    }
+
+    status = prepareInputsAndModelForInference();
+    if (!status.ok()) {
+        return status;
+    }
+    auto& inferRequestsQueue = this->model->getInferRequestsQueue();
+    this->nodeStreamIdGuard = std::make_unique<NodeStreamIdGuard>(inferRequestsQueue);
+    return status;
+}
+
+Status DLNodeSession::prepareInputsAndModelForInference() {
+    size_t requestedBatchSize = 0;
+    std::map<std::string, shape_t> requestedReshapes;
+
+    // Validate each blob against its OV tensor info
+    const auto& inputsInfo = this->model->getInputsInfo();
+    for (const auto& kv : this->inputHandler->getInputs()) {
+        const auto& name = kv.first;
+        auto& blob = kv.second;
+
+        if (inputsInfo.count(name) == 0) {
+            std::stringstream ss;
+            ss << "Required input: " << name;
+            const std::string details = ss.str();
+            SPDLOG_LOGGER_DEBUG(dag_executor_logger, "[Node: {}] Missing input with specific name - {}", getName(), details);
+            return Status(StatusCode::INVALID_MISSING_INPUT, details);
+        }
+        auto& inputInfo = *inputsInfo.at(name);
+        auto status = validate(blob, inputInfo);
+        if (status.ok()) {
+            continue;
+        }
+
+        // If precision is incorrect, perform conversion
+        if (status == StatusCode::INVALID_PRECISION) {
+            return status;
+        }
+
+        // If batch size is incorrect, perform network batch size change if allowed (shape mode=auto or batch size=auto)
+        if (status == StatusCode::INVALID_BATCH_SIZE) {
+            if (this->model->getModelConfig().getBatchingMode() == Mode::AUTO) {
```

Comment:
We do not support dynamic batch and shape, but we still have this code.

Reply:
I just did the move of code from dl_node.cpp. DId not want to introduce additional changes at this point - especially that still some unit tests use this functionality.